docs/pages/pmd/devdocs/major_contributions/adding_new_language.md

```diff
@@ -87,3 +87,13 @@ folder: pmd/devdocs
     This will load all rulesets and verify, that all required attributes are provided.
 
     *Note:* You'll need to add your ruleset to `rulesets.properties`, so that it can be found.
+
+## Debugging with Rule Designer
+
+When implementing your grammar it may be very useful to see how PMD parses your example files.
+This can be achieved with Rule Designer:
+*   Take some [fresh Rule Designer build](https://github.com/pmd/pmd-designer) and unpack the zip archive. It should contain `bin` and `lib` subdirectories.
```

Comment:
Which zip-archive to you mean? Do you mean the normal pmd binary distribution? Then the URL would be https://github.com/pmd/pmd/releases and the file to download is "pmd-bin-<version>.zip". The PMD binary distribution contains the rule designer.

Reply:
Oops! Do I get it right, it is merely built together with PMD from the main `pmd` repository. Then this step is not required at all and can be harmful (with respect to some strange bugs because of version mismatch). I just thought that since it resides in a separate repo, it is built separately, but then checked build artifact, and looks like Designer is there too :)

Reply:
It built seperately nowadays, but we ship the latest released version of the designer with the main PMD build. This should provide you a working combination of the two tools.

You can then e.g. update the designer by replacing the pmd-designer-x.y.z.jar in the lib folder, as described on the readme of the designer: https://github.com/pmd/pmd-designer/#updating
We try to keep them compatible version-wise, but bugs might happen.

The step would only be needed, if there is a newer pmd-designer version (like a self built latest snapshot), that is not yet shipped with the pmd-binary.